### DISCUSSION TITLE
Trying conversion not working

### DISCUSSION THREAD

**Discussion by mtnmanofGod** [apprenticeharper/DeDRM_tools#1505] [Q&A]
Hi Guys! Newbe here so please forgive my ignorance. I added the deDRM 7.0.3 plugin to Calibre 5.0 successfully by all indications but when I tried to do the convert to PDF it just gave me the usual error that it is  protected?? Am I doing something wrong? I have a screen shot but I cannot attach it here apparently? Thank You kindly for your time!

**Comment by ElleKayEm** [comment]
DeDRM 7.0.3 still has some problems with PDFs.  Try with calibre 4.23 and DeDRM 6.8.1. If that doesn't do it, post a log:  https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Reply by mtnmanofGod** [reply]
HERE IS THE ERROR LOG I RECEIVED

calibre, version 4.23.0 (win32, embedded-python: True)
Conversion error: Failed: Convert book 1 of 1 (The Blessing)

Convert book 1 of 1 (The Blessing)
Conversion options changed from defaults:
  read_metadata_from_opf: u'C:\\Users\\deute\\AppData\\Local\\Temp\\calibre_chjewx\\54dfcr.opf'
  pdf_serif_family: u'MS Shell Dlg 2'
  output_profile: u'generic_eink'
  verbose: 2
  cover: u'C:\\Users\\deute\\AppData\\Local\\Temp\\calibre_chjewx\\bqbq01.jpeg'
  pdf_sans_family: u'MS Shell Dlg 2'
Resolved conversion options
calibre version: 4.23.0
{'asciiize': False,
 'author_sort': None,
 'authors': None,
 'base_font_size': 0.0,
 'book_producer': None,
 'change_justification': u'original',
 'chapter': u"//*[((name()='h1' or name()='h2') and re:test(., '\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', 'i')) or @class = 'chapter']",
 'chapter_mark': u'pagebreak',
 'comments': None,
 'cover': u'C:\\Users\\deute\\AppData\\Local\\Temp\\calibre_chjewx\\bqbq01.jpeg',
 'custom_size': None,
 'debug_pipeline': None,
 'dehyphenate': True,
 'delete_blank_paragraphs': True,
 'disable_font_rescaling': False,
 'duplicate_links_in_toc': False,
 'embed_all_fonts': False,
 'embed_font_family': None,
 'enable_heuristics': False,
 'expand_css': False,
 'extra_css': None,
 'filter_css': u'',
 'fix_indents': True,
 'font_size_mapping': None,
 'format_scene_breaks': True,
 'html_unwrap_factor': 0.4,
 'input_encoding': None,
 'input_profile': <calibre.customize.profiles.InputProfile object at 0x000001D7A55CDD88>,
 'insert_blank_line': False,
 'insert_blank_line_size': 0.5,
 'insert_metadata': False,
 'isbn': None,
 'italicize_common_cases': True,
 'keep_ligatures': False,
 'language': None,
 'level1_toc': None,
 'level2_toc': None,
 'level3_toc': None,
 'line_height': 0.0,
 'linearize_tables': False,
 'margin_bottom': 5.0,
 'margin_left': 5.0,
 'margin_right': 5.0,
 'margin_top': 5.0,
 'markup_chapter_headings': True,
 'max_toc_links': 50,
 'minimum_line_height': 120.0,
 'no_chapters_in_toc': False,
 'no_inline_navbars': False,
 'output_profile': <calibre.customize.profiles.GenericEink object at 0x000001D7A55E39C8>,
 'page_breaks_before': u"//*[name()='h1' or name()='h2']",
 'paper_size': u'letter',
 'pdf_add_toc': False,
 'pdf_default_font_size': 20,
 'pdf_footer_template': None,
 'pdf_header_template': None,
 'pdf_hyphenate': False,
 'pdf_mark_links': False,
 'pdf_mono_family': u'Courier',
 'pdf_mono_font_size': 16,
 'pdf_odd_even_offset': 0.0,
 'pdf_page_margin_bottom': 72.0,
 'pdf_page_margin_left': 72.0,
 'pdf_page_margin_right': 72.0,
 'pdf_page_margin_top': 72.0,
 'pdf_page_number_map': None,
 'pdf_page_numbers': False,
 'pdf_sans_family': u'MS Shell Dlg 2',
 'pdf_serif_family': u'MS Shell Dlg 2',
 'pdf_standard_font': u'serif',
 'pdf_use_document_margins': False,
 'prefer_metadata_cover': False,
 'preserve_cover_aspect_ratio': False,
 'pretty_print': False,
 'pubdate': None,
 'publisher': None,
 'rating': None,
 'read_metadata_from_opf': u'C:\\Users\\deute\\AppData\\Local\\Temp\\calibre_chjewx\\54dfcr.opf',
 'remove_fake_margins': True,
 'remove_first_image': False,
 'remove_paragraph_spacing': False,
 'remove_paragraph_spacing_indent_size': 1.5,
 'renumber_headings': True,
 'replace_scene_breaks': u'',
 'search_replace': '[]',
 'series': None,
 'series_index': None,
 'smarten_punctuation': False,
 'sr1_replace': None,
 'sr1_search': None,
 'sr2_replace': None,
 'sr2_search': None,
 'sr3_replace': None,
 'sr3_search': None,
 'start_reading_at': None,
 'subset_embedded_fonts': False,
 'tags': None,
 'timestamp': None,
 'title': None,
 'title_sort': None,
 'toc_filter': None,
 'toc_threshold': 6,
 'toc_title': None,
 'transform_css_rules': '[]',
 'uncompressed_pdf': False,
 'unit': u'inch',
 'unsmarten_punctuation': False,
 'unwrap_lines': True,
 'use_auto_toc': False,
 'use_profile_size': False,
 'verbose': 2}
InputFormatPlugin: MOBI Input running
on C:\Users\deute\AppData\Local\Temp\calibre_chjewx\x3zpyz.azw3
Found KF8 MOBI of type u'standalone'
Python function terminated unexpectedly
  The Blessing (Error Code: 1)
 Traceback (most recent call last):
  File "site.py", line 114, in main
  File "site.py", line 88, in run_entry_point
  File "site-packages\calibre\utils\ipc\worker.py", line 209, in main
  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 43, in gui_convert_override
  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 28, in gui_convert
  File "site-packages\calibre\ebooks\conversion\plumber.py", line 1110, in run
  File "site-packages\calibre\customize\conversion.py", line 246, in __call__
  File "site-packages\calibre\ebooks\conversion\plugins\mobi_input.py", line 47, in convert
  File "site-packages\calibre\ebooks\mobi\reader\mobi8.py", line 86, in __call__
  File "site-packages\calibre\ebooks\mobi\reader\mobi6.py", line 158, in check_for_drm
DRMError: The Blessing

**Reply by ElleKayEm** [reply]
Don't see DeDRM trying to do anything here.  Follow the instructions at the link I gave to post an import log.

**Reply by mtnmanofGod** [reply]
calibre Debug log
calibre 4.23 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 2.7.16+
Windows: ('10', '10.0.19041', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0)
calibre 4.23 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 2.7.16+
Windows: ('10', '10.0.19041', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 127.664921466 x 127.590697674
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.42] splash screen shown
[0.42] Initializing db...
[0.42] db initialized
[0.42] Constructing main UI...
[1.17] main UI initialized...
[1.17] Hiding splash screen
[3.92] splash screen hidden
[3.92] Started up in 3.92 seconds with 25 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['\xa6\x8aA\x8b\xf8\xea\x9bF\xdco\x91@\xfdJ<\xb2\x16\xda\x8e{\xb4J,w\xa5\xa2\x04\xe1\xdb7\xdc\xa2', 'D:\\Google Drive\\A Reformed Theological Library\\The Blessing\\The Blessing.azw3']
DeDRM v6.8.0: Trying to decrypt The Blessing.azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: The Blessing
Found 0 keys to try after 0.0 seconds
Crypto Type is: 2
DeDRM v6.8.0: Failed to decrypt with error: No key found in 0 keys tried.
DeDRM v6.8.0: Looking for new default Kindle Key after 0.0 seconds
searching for kinfoFiles in C:\Users\deute\AppData\Local
Found K4PC 1.25+ kinf2018 file: C:\Users\deute\AppData\Local\Amazon\Kindle\storage\.kinf2018
Decrypted key file using IDString '79659746' and UserName '6465757465'
DeDRM v6.8.0: Found 1 new key
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: The Blessing
Got DSN key from database default_key_1
Found 4 keys to try after 5.8 seconds
Crypto Type is: 2
DeDRM v6.8.0: Ultimately failed to decrypt after 5.8 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 626, in run
  File "calibre_plugins.dedrm.__init__", line 576, in KindleMobiDecrypt
DeDRMError: DeDRM v6.8.0: Ultimately failed to decrypt after 5.8 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added The Blessing to db in: 1.7
Added 1 books in 8.0 seconds

**Reply by ElleKayEm** [reply]
Just realized I misread your first post.  I was thinking you were trying to DeDRM a PDF.  Anyway, if that's from or for an E-ink Kindle (not a Fire), you need to add the serial number to DeDRM's customization.

**Reply by mtnmanofGod** [reply]
I'm on a PC and I downloaded the book to my PC to DeDRM the file using my Kindle as the target device using the download to USB drive option from Amazon download screen. I did not know that each book downloaded was device specific in the code. Should I have used the copy of the book that's downloaded into my PC Kindle app, is that possible? Because I don't have my Kindle anymore, I use my PC all the time.

**Reply by ElleKayEm** [reply]
Is your Kindle still on your Amazon account?  Copy the serial number from your content and devices page.

Otherwise, yes, you can use books downloaded with Kindle for PC instead.  Needs to be on the same machine and user as calibre with DeDRM.  See this post about getting books you can decrypt:  https://www.mobileread.com/forums/showthread.php?t=283371

**Reply by mtnmanofGod** [reply]
Elle, Where do I put the serial # for the Kindle in the customization DeDRM?

**Reply by ElleKayEm** [reply]
Preferences > Plugins.  Select DeDRM in the list, then Customize plugin > eInk Kindle ebooks > green plus sign.  Once you've entered it, be sure to click OK and Apply on your way back out of the dialog boxes.  Then restart calibre before adding your book.

**Comment by mtnmanofGod** [comment]
Elle ... I DID IT! I COULDN'T HAVE DONE IT WITHOUT YOU! THANK YOU SO MUCH
FOR TAKING THE TIME TO HELP ME ON THIS. DRM is a tool of arbitrary control
that presumes the wrong use of a book for republication on a mass scale. We
should have the same access to the material as we do with a paper book. Good Work guys!

**Comment by wsblk123** [comment]
I am very new to this. I purchased a book and now it keeps on saying, the content you're trying to read need the latest version. i had  1.17. 
I cant even find where the books are located stored on my windows machine  - i want to convert this to a mobi.  can someone give me a lead?  

This book had the tag _"On PC & Mac, the latest version of the Kindle app is required"_ and i missed it.

**Reply by ElleKayEm** [reply]
Sounds like the book is only available in KFX format.  If you can download it with Kindle for PC 1.26, then you'll be able to decrypt it.  You'll also need the KFX Input plugin.  Books are located in the Kindle Content folder in your Documents folder.  See method 4 at this post:  https://www.mobileread.com/forums/showthread.php?t=283371

**Reply by wsblk123** [reply]
Thank you, i tried using 1.26 no it doesn't work, I get the message **your kindle app requires an update to view this content Click here to downlead the latest version**  and in the downloads folder I can find azw.md file - which should be the book. Please help

**Reply by ElleKayEm** [reply]
In that case, you won't be able to decrypt it at this time.

**Reply by wsblk123** [reply]
That's sad. I guess it's the newest version.  hope someone has some idea, they can help me with
